### PR TITLE
Additional header validations

### PR DESCRIPTION
**Pull request by Nemanja0x**
# Description

The PR add more validations for header in validationHeaderFields.

Check header ExtraData length. If length is less than 32 bytes return error.
Check header Nonce equals to zero, if not return error.
Check header GasUsed less than headers GasiLimit, if exceed return error.

# Changes include

- [ ] Bugfix (non-breaking change that solves an issue)
- [ ] Hotfix (change that solves an urgent issue, and requires immediate attention)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Breaking changes

Please complete this section if any breaking changes have been made, otherwise delete it

# Checklist

- [x] I have assigned this PR to myself
- [x] I have added at least 1 reviewer
- [ ] I have added the relevant labels
- [ ] I have updated the official documentation
- [ ] I have added sufficient documentation in code

## Testing

- [x] I have tested this code with the official test suite
- [ ] I have tested this code manually

### Manual tests

Please complete this section if you ran manual tests for this functionality, otherwise delete it

# Documentation update

Please link the documentation update PR in this section if it's present, otherwise delete it

# Additional comments

Please post additional comments in this section if you have them, otherwise delete it

### CODE REVIEW

File: consensus/polybft/fsm.go [hunk 1]
```diff
@@ -632,6 +633,10 @@ func (f *fsm) verifyDistributeRewardsTx(distributeRewardsTx *types.Transaction)
 }
 
 func validateHeaderFields(parent *types.Header, header *types.Header) error {
+	// header extra data must be higher or equal to ExtraVanity = 32 in order to be complient with Ethereum blocks
```
Comment (Stefan-Ethernal) [inline consensus/polybft/fsm.go]:
```suggestion
	// header extra data must be higher or equal to ExtraVanity = 32 in order to be compliant with Ethereum blocks
```

File: consensus/polybft/fsm.go [hunk 2]
```diff
@@ -43,6 +43,7 @@ var (
 		"is either nil or it does not match the received one")
 	errValidatorSetDeltaMismatch        = errors.New("validator set delta mismatch")
 	errValidatorsUpdateInNonEpochEnding = errors.New("trying to update validator set in a non epoch ending block")
+	zeroNonce                           = types.Nonce{}
```
Comment (Stefan-Ethernal) [inline consensus/polybft/fsm.go]: Maybe put this into `types.go` and reuse it from there?

Comment (Nemanja0x) [reply consensus/polybft/fsm.go]: fixed: https://github.com/0xPolygon/polygon-edge/pull/1508/commits/2dbdab21c1a4f5e8d850222fce96e40590deb492

File: types/header.go [hunk 3]
```diff
@@ -53,6 +53,10 @@ func (h *Header) IsGenesis() bool {
 
 type Nonce [8]byte
 
+func (n Nonce) EncodeNonce(i uint64) {
+	binary.BigEndian.PutUint64(n[:], i)
+}
```
Comment (Stefan-Ethernal) [inline types/header.go]: Optional nitpick:
Not sure do we need this function anymore? In the original PR submitted by Dušan, it was used in multiple places, so that's why I've suggested extracting that logic in a new function.

**Review by Stefan-Ethernal — APPROVED** [review]
LGTM

**Review by goran-ethernal — APPROVED** [review]
LGTM, but maybe consider adding couple of UTs that test these negative paths.

File: types/header.go (hunk 3 above)
Comment (Nemanja0x) [reply types/header.go]: fixed: https://github.com/0xPolygon/polygon-edge/pull/1508/commits/76b225ee32a8cfad17b9a671d6fd0a61654f8f9f

**Comment by Nemanja0x** [comment]
> LGTM, but maybe consider adding couple of UTs that test these negative paths.

fixed: https://github.com/0xPolygon/polygon-edge/pull/1508/commits/76b225ee32a8cfad17b9a671d6fd0a61654f8f9f
